test/bdd/pkg/route/route_rest_steps.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package route
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/cucumber/godog"
+
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
+	"github.com/hyperledger/aries-framework-go/test/bdd/pkg/context"
+)
+
+var logger = log.New("aries-framework/tests/messaging")
+
+type registerRouteReq struct {
+	ConnectionID string `json:"connectionID"`
+}
+
+// RESTSteps is steps for route using REST APIs
+type RESTSteps struct {
+	bddContext *context.BDDContext
+}
+
+// NewRouteRESTSteps return steps for route using REST APIs
+func NewRouteRESTSteps(ctx *context.BDDContext) *RESTSteps {
+	return &RESTSteps{
+		bddContext: ctx,
+	}
+}
+
+// RegisterRoute registers the router for the agent.
+func (d *RESTSteps) RegisterRoute(agentID, varName string) error {
+	connectionID := d.bddContext.Args[varName]
+
+	fmt.Println("connectionID == ", connectionID)
+
+	destination, ok := d.bddContext.GetControllerURL(agentID)
+	if !ok {
+		return fmt.Errorf(" unable to find controller URL registered for agent [%s]", agentID)
+	}
+
+	err := sendHTTP(http.MethodPost, destination+"/route/register", registerRouteReq{ConnectionID: connectionID}, nil)
+	if err != nil {
+		return fmt.Errorf("failed to handle invitation: %w", err)
+	}
+
+	return nil
+}
+
+// UnregisterRoute unregisters the router.
+func (d *RESTSteps) UnregisterRoute(agentID string) error {
+	destination, ok := d.bddContext.GetControllerURL(agentID)
+	if !ok {
+		return fmt.Errorf(" unable to find controller URL registered for agent [%s]", agentID)
+	}
+
+	err := sendHTTP(http.MethodDelete, destination+"/route/unregister", nil, nil)
+	if err != nil {
+		// ignore error if router is not registered (code=5003)
+		if strings.Contains(err.Error(), "\"code\":5003") {
+			logger.Infof("ignore unregister - router not registered")
+
+			return nil
+		}
+
+		return fmt.Errorf("failed to handle invitation: %w", err)
+	}
+
+	return nil
+}
+
+// RegisterSteps registers router steps
+func (d *RESTSteps) RegisterSteps(s *godog.Suite) {
+	s.Step(`^"([^"]*)" sets connection "([^"]*)" as the router$`, d.RegisterRoute)
```

Comment:
it might collide with this, double check https://github.com/hyperledger/aries-framework-go/blob/master/test/bdd/pkg/route/route_sdk_steps.go#L56